src/components/postCard/children/upvoteButton.tsx

```diff
@@ -0,0 +1,160 @@
+    content: any,
+    isVoting: boolean,
+    activeVotes: any[],
+    isShowPayoutValue?: boolean,
+    boldPayout?: boolean,
+    parentType?: PostTypes;
+    onUpvotePress: (anchorRect: Rect) => void,
+    onPayoutDetailsPress: (anchorRef: Rect) => void,
+}
+
+export const UpvoteButton = ({
+    content,
+    isVoting,
+    activeVotes,
+    isShowPayoutValue,
+    boldPayout,
+    onUpvotePress,
+    onPayoutDetailsPress
+}: UpvoteButtonProps) => {
+
+    const upvoteRef = useRef(null);
+    const detailsRef = useRef(null);
+
+    const currentAccount = useAppSelector((state => state.account.currentAccount));
+
+
+    const [isVoted, setIsVoted] = useState(null);
+    const [isDownVoted, setIsDownVoted] = useState(null);
+
+
+
+    useEffect(() => {
+        let _isMounted = true;
+
+        const _calculateVoteStatus = async () => {
+
+            //TODO: do this heavy lifting during parsing or react-query/cache response
+            const _isVoted = await isVotedFunc(activeVotes, currentAccount?.name);
+            const _isDownVoted = await isDownVotedFunc(activeVotes, currentAccount?.name);
+
+            if (_isMounted) {
+                setIsVoted(_isVoted && parseInt(_isVoted, 10) / 10000);
+                setIsDownVoted(_isDownVoted && (parseInt(_isDownVoted, 10) / 10000) * -1);
+            }
+        };
+
+        _calculateVoteStatus();
+        return () => { _isMounted = false };
+    }, [activeVotes]);
+
+
+    const _getRectFromRef = (ref: any, callback: (anchorRect: Rect) => void) => {
+        const handle = findNodeHandle(ref.current);
+        if (handle) {
+            NativeModules.UIManager.measure(handle, (x0, y0, width, height, x, y) => {
+                const anchorRect: Rect = { x, y, width, height };
+                callback(anchorRect)
+            });
+        }
+    }
+
+
+    const _onPress = () => {
+        _getRectFromRef(upvoteRef, onUpvotePress);
+    }
+
+    const _onDetailsPress = () => {
+        _getRectFromRef(detailsRef, onPayoutDetailsPress)
+    }
+
+
+    const isDeclinedPayout = content?.is_declined_payout;
+    const totalPayout = content?.total_payout;
+    const maxPayout = content?.max_payout;
+
+    const payoutLimitHit = totalPayout >= maxPayout;
+    const _shownPayout = payoutLimitHit && maxPayout > 0 ? maxPayout : totalPayout;
+
+
+
+
+    let iconName = 'upcircleo';
+    const iconType = 'AntDesign';
+    let downVoteIconName = 'downcircleo';
+
+    if (isVoted) {
+        iconName = 'upcircle';
+    }
+
+    if (isDownVoted) {
+        downVoteIconName = 'downcircle';
+    }
+
+
+
+    return (
+        <View style={styles.container}>
+            <TouchableOpacity
+                ref={upvoteRef}
+                onPress={_onPress}
+                style={styles.upvoteButton}
+            >
+                <Fragment>
+                    {isVoting ? (
+                        <View style={{ width: 19 }}>
+                            <PulseAnimation
```

Comment:
We added new animation to heart icon after vote is cast we animated heart as well on website. Good to have it on app as well, in separate task of course, just mentioning so we don't forget. Check it out please!

Reply:
I rather removed pulse animation in latest code in favour of using PENDING voting state in cache, will check out heart animation as well, adding that as separate task in backlog

Reply:
We did keep pulse animation on vote button and added new animation to heart icon (number of votes) icon. That's extra animation. Removing pulse now is not required, imo.